### PR TITLE
Backend docker support

### PR DESCRIPTION
**Pull request by LeaX-XIV**

### CODE REVIEW

File: server/functions.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		if(!isset($_SESSION["user_id"]) || !isset($_SESSION["nonce"])){
+			return false;
+		}
+
+		$stmt = $pdo->prepare("SELECT ID, username, password, type FROM users WHERE ID = :userId");
+		$stmt->bindValue(":userId", $_SESSION["user_id"], PDO::PARAM_INT);
+
+
+		if (!$stmt->execute()) {
+			throw new Error($stmt->errorInfo(), $stmt->errorCode());
```
Comment (LeaX-XIV) [inline server/functions.php]: Type error
```suggestion
			throw new PDOException($stmt->errorInfo()[2], $stmt->errorInfo()[0]);
```

File: server/scheduled/send_notifications.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		$db = new PDO("../db.sqlite");
+
+		$stmt = $db->prepare("SELECT users.email, users.firstname, users.lastname, courses.name AS course_name, COUNT(*) AS booking_num FROM users, lectures, courses, bookings WHERE
+			users.id = courses.teacher_id AND courses.ID = lectures.course_id AND lectures.ID = bookings.lecture_id AND bookings.cancellation_ts IS NULL AND
+			lectures.start_ts >= :startTs AND lectures.start_ts < :endTs");
+		$stmt->bindValue(":startTs", $start_time->getTimestamp());
+		$stmt->bindValue(":endTs", $end_time->getTimestamp());
+
+		if (!$stmt->execute()) {
+			throw new PDOException($stmt->errorInfo(), $stmt->errorCode());
```
Comment (LeaX-XIV) [inline server/scheduled/send_notifications.php:24]: Type error
```suggestion
			throw new PDOException($stmt->errorInfo()[2], $stmt->errorInfo()[0]);
```

File: server/scheduled/send_notifications.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		}
+
+		while($data = $stmt->fetch()){
+			mail($data["email"], "Notification of students booked for a lesson", "Good morning ".$data["firstname"]." ".$data["lastname"].",\nFor your lecture of ".$data["course_name"]." there are ".$data["booking_num"]." active bookings.\nKind Regards");
```
Comment (LeaX-XIV) [inline server/scheduled/send_notifications.php]: More meaningful message.
Should also mention the time and room of the lecture.
```suggestion
			mail($data["email"],
			'[' . $data['course_name'] . '] Booking Notice',
			"Dear ".$data["lastname"]." ".$data["firstname"].",\n" . $data["booking_num"] . " have booked for the lecture of ".$data["course_name"].".\nKind Regards");
```

File: server/scheduled/send_notifications.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		$end_time = clone $start_time;
+		$end_time->modify(("+1 day"));
+
+		$db = new PDO("../db.sqlite");
+
+		$stmt = $db->prepare("SELECT users.email, users.firstname, users.lastname, courses.name AS course_name, COUNT(*) AS booking_num FROM users, lectures, courses, bookings WHERE
+			users.id = courses.teacher_id AND courses.ID = lectures.course_id AND lectures.ID = bookings.lecture_id AND bookings.cancellation_ts IS NULL AND
+			lectures.start_ts >= :startTs AND lectures.start_ts < :endTs");
+		$stmt->bindValue(":startTs", $start_time->getTimestamp());
+		$stmt->bindValue(":endTs", $end_time->getTimestamp());
```
Comment (LeaX-XIV) [inline server/scheduled/send_notifications.php:21]: Check that the lecture is in presence and not cancelled.
```suggestion
		$stmt = $db->prepare("SELECT users.email, users.firstname, users.lastname, courses.name AS course_name, COUNT(*) AS booking_num FROM users, lectures, courses, bookings WHERE
			users.id = courses.teacher_id AND courses.ID = lectures.course_id AND lectures.ID = bookings.lecture_id AND bookings.cancellation_ts IS NULL AND
			lectures.start_ts >= :startTs AND lectures.start_ts < :endTs AND settings & :lectureRemote & :lectureCancelled = 0");
		$stmt->bindValue(":startTs", $start_time->getTimestamp());
		$stmt->bindValue(":endTs", $end_time->getTimestamp());
		$stmt->bindValue(":lectureRemote", LECTURE_REMOTE);
		$stmt->bindValue(":lectureCancelled", LECTURE_CANCELLED);
```

File: server/scheduled/send_notifications.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+	case "send_booking_number_notifications":
+		$is_web=http_response_code()!==FALSE;
+
+		if(!$is_web) die();
+
+		$start_time = new DateTime('tomorrow', new DateTimeZone($server_default_timezone));
+		$end_time = clone $start_time;
+		$end_time->modify(("+1 day"));
+
+		$db = new PDO("../db.sqlite");
```
Comment (LeaX-XIV) [inline server/scheduled/send_notifications.php]:
```suggestion
		$db = new PDO("sqlite:../db.sqlite");
```

File: server/scheduled/send_notifications.php [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+switch($argv[1]){
+	case "send_booking_number_notifications":
+		$is_web=http_response_code()!==FALSE;
+
+		if(!$is_web) die();
+
+		$start_time = new DateTime('tomorrow', new DateTimeZone($server_default_timezone));
+		$end_time = clone $start_time;
+		$end_time->modify(("+1 day"));
```
Comment (LeaX-XIV) [inline server/scheduled/send_notifications.php:13]: I know that ther will not be lectures starting at midnight, but technically it is more correct. Check if modify accepts 2 modifiers in the same string.
```suggestion
		$end_time->modify(("+1 day - 1 second"));
```

Comment (Francexi) [reply server/scheduled/send_notifications.php:13]: i'm sure it does. For now leave as is it

File: server/scheduled/send_notifications.php (hunk 3 above)
Comment (Francexi) [reply server/scheduled/send_notifications.php]: I'll add time and room if I have time to do that before the demo of tomorrow. Implement missing stories and testing them has precedence

**Pull request merged**